### PR TITLE
Video upload improvements

### PR DESCRIPTION
**Pull request by mneuwert**
## Description
- Add ability to upload slo-mo videos etc
- Add option to allow uploading original videos

## Related Issue
#847 

## Motivation and Context
More flexibility for photo uploads

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING**](https://github.com/owncloud/ios-app/blob/master/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by michaelstingl** [comment]
> * Add ability to upload slo-mo videos etc

Looks very good 👍 

CleanShot 2020-12-23 at 01 38 59

**Comment by michaelstingl** [comment]
Hm, interesting. Apple's new 10bit Dolby Vision format doesn't get re-encoded at all:

CleanShot 2020-12-23 at 01 52 11

**Comment by michaelstingl** [comment]
I found, "Prefer original videos" is only respected when upload is initiated from inside the app. Video uploads initiated from Photos app via Share Sheet or via FileProvider always have "Preparing" phase before upload, so they get re-encoded. Add a note under the switch?

**Comment by mneuwert** [comment]
@michaelstingl hm.. actually these shall be uploaded as is without any processing. It can be any binary file not just video, right?

**Comment by michaelstingl** [comment]
Check Settings > Media upload:

“Prefer original videos” and “Prefer RAW photos” seems connected. Enabling the one, also enabled the other. Disabling results in same behavior:


https://user-images.githubusercontent.com/214010/103462625-d0ac9780-4d26-11eb-86b1-41ff39c67d72.MOV

**Comment by mneuwert** [comment]
> Check Settings > Media upload:
> 
> “Prefer original videos” and “Prefer RAW photos” seems connected. Enabling the one, also enabled the other. Disabling results in same behavior:
> 
>  B6FD2EC3-ECAC-49D2-85FF-D78277CC2082.MOV

This is fixed with latest commit

File: fastlane/Fastfile [hunk 1]
```diff
@@ -613,14 +613,14 @@ end
...
-      add_badge(custom: "fastlane/badge.png")
-    end
+    #if values[:BETA_APP_ICON]
+    #  commit = last_git_commit
+    #  short_hash = commit[:abbreviated_commit_hash] # short sha of commit
+    #  sh "brew install librsvg"
+    #  sh "sed -e \"s/\#version#/" + version + "/\" -e \"s/\#githash#/" + short_hash + "/\" badge.svg > badge_tmp.svg"
+    #  sh "rsvg-convert badge_tmp.svg > badge.png"
+    #  add_badge(custom: "fastlane/badge.png")
+    #end
```
Comment (hosy) [inline fastlane/Fastfile:623]: This needs to be uncommented

File: ownCloud/Settings/ProPhotoUploadSettingsSection.swift [hunk 2]
```diff
@@ -98,14 +109,20 @@ class ProPhotoUploadSettingsSection: SettingsSection {
 
 		self.add(row: preferOriginalsRow)
 
-		if AVCaptureDevice.rawCameraDeviceAvailable() {
```
Comment (hosy) [inline ownCloud/Settings/ProPhotoUploadSettingsSection.swift:101]: Why did you remove the `if` clause? Raw camera support is not available on every device.

**Review by hosy — CHANGES_REQUESTED** [review]

File: ownCloud/Settings/ProPhotoUploadSettingsSection.swift (hunk 2 above)
Comment (michaelstingl) [reply ownCloud/Settings/ProPhotoUploadSettingsSection.swift:101]: > Why did you remove the `if` clause? Raw camera support is not available on every device.

What about raw files stored on devices without the latest camera? I created the file in iPhone 12 Pro Max, but did my upload testing with the iPhone 8 test device.

Comment (hosy) [reply ownCloud/Settings/ProPhotoUploadSettingsSection.swift:101]: Ok, good point. I missed that option!

File: fastlane/Fastfile (hunk 1 above)
Comment (felix-schwarz) [reply fastlane/Fastfile:623]: I commented that out because it failed and prevented Bitrise builds a few weeks ago (IIRC). When putting it back in, please check whether IPAs are still being built.

Comment (hosy) [reply fastlane/Fastfile:623]: Should be fixed with the latest `master` merged inside

**Review by felix-schwarz — APPROVED** [review]

File: ownCloud/Settings/ProPhotoUploadSettingsSection.swift (hunk 2 above)
Comment (mneuwert) [reply ownCloud/Settings/ProPhotoUploadSettingsSection.swift:101]: @hosy that was exactly the intension to allow users upload RAW photos from their iCloud synced photo library shot on more capable device.

**Comment by jesmrec** [comment]
Checked the slo-mo videos, they are correctly uploaded and played. @michaelstingl collected all relevant encoding info above. 

About uploading original videos:

- [x] Settings option works fine. It is only visible in enterprise version servers, and under `Prefer original videos` IAP feature
- [x] Option enabled + video edited = upload video without the edit. 
- [x] Option disabled + video edited = upload video with the edit.
- [x] An original video is always correctly uploaded, no matter the Setting value

About encoding, i will add a report about the edited video with and without the Setting enabled. For me it's ok, but you can take a look if any parameter is more relevant

| Original | Edited |
| :-: | :-: |
|Screenshot 2021-01-15 at 11 13 59|Screenshot 2021-01-15 at 11 14 21|

**Comment by jesmrec** [comment]
Approved

**Review by jesmrec — APPROVED** [review]
